### DISCUSSION TITLE
unity模型文件外置替换

### DISCUSSION THREAD

**Discussion by StarDamnSucks** [alex8088/electron-vite#182] [Q&A]
目前我使用electron-vite遇到两个问题
（1）在开发模式下如何访问resources目录下的文件？我需要在组件中使用unity，由于unity模型需要可以进行替换，因此我将unity模型文件放在resouces文件夹下，我目前使用“../../resources/”的路径可以在打包后访问到文件，但是在开发模式下不能访问到，这个问题怎么解决？
（2）由于项目中的unity文件需要可替换，因此我放在了resources文件夹下，但是尽管目前能够正确加载，但是当unity模型文件在替换后就不能正常运行了，似乎是因为loader.js文件被打包进了asar中，放在public文件夹下可以正常运行，但是在打包后相关资源并没有放入resources文件夹下，因此unity的模型文件不能替换
我找了一圈没有找到解决办法，希望能够帮我解答下疑问，感谢

**Comment by alex8088** [comment]
如何引用untiy文件

**Reply by StarDamnSucks** [reply]
我目前使用的是unity-webgl这个组件，通过设定loaderUrl属性的方式使用，引用的路径是“../../resources/xxx.loader.js”，文件在resources文件夹下，我在文档里看到说渲染进程的public文件夹下的目录会被复制到输出目录，在我打包后根据控制台信息看到会将loader.js文件打包进asar包中，因此就不能在外部对模型文件进行替换

**Reply by alex8088** [reply]
> 我目前使用的是unity-webgl这个组件，通过设定loaderUrl属性的方式使用，引用的路径是“../../resources/xxx.loader.js”，文件在resources文件夹下，我在文档里看到说渲染进程的public文件夹下的目录会被复制到输出目录，在我打包后根据控制台信息看到会将loader.js文件打包进asar包中，因此就不能在外部对模型文件进行替换

你需要区分这些资源是给主进程还是渲染进程使用
- resources 是专用于主进程和预加载脚本的公共目录，他是不会被复制到输出目录的，打包工具设定不打包到asar中。
- 如果给渲染进程使用，public文件夹的文件会被复制到输出目录，打包工具是没有默认设置排除到asar外

明确如果不是import方式导入，也可以放在resources 中， 正确处理路径就好（但注意打包后路径是有变化的 [app.asar.unpacked 文件路径问题](https://cn-evite.netlify.app/guide/assets.html#app-asar-unpacked-%E6%96%87%E4%BB%B6%E8%B7%AF%E5%BE%84%E9%97%AE%E9%A2%98)）

如果通过import方式导入，你可以设置renderer 的公共目录在root下的任意目录，同时指定不复制公共目录（详细参考vite配置文档），并在electorn-builder中配置不将目录打包到asar。需要注意的是，这会对renderer的其他静态资源处理产生影响。
